frontend/src/components/ViewActivities/activityfns.js

```diff
@@ -54,4 +58,32 @@ export function getField(activity, fieldName, defaultValue){
     return defaultValue;
   }
   return activity[fieldName];
+}
+
+/**
+ * Write contents into an activity in the database.
+ * 
+ * @param {string} tripId Database ID of the trip whose actiivty should be modified.
+ * @param {string} activityId Database ID of the activity to be modified.
+ * @param {Object} newValues Dictionary of the new values in {fieldName: newValue} form
+ * @returns true if the write was successful, false otherwise.
+ */
+export async function writeActivity(tripId, activityId, newValues) {
+  // todo: check if tripId or activityId is not valid. (#58)
+  newValues = {
+    ...newValues, 
+    'fillerstamp': Firebase.firestore.Timestamp.now()
+  };
+
+  const act = db.collection(DB.COLLECTION_TRIPS).doc(tripId)
```

Comment:
Nit: call this 'activity' instead?
```suggestion
  const activity = db.collection(DB.COLLECTION_TRIPS).doc(tripId)
```